storm-core/src/jvm/org/apache/storm/blobstore/BlobStore.java

```diff
@@ -128,6 +128,15 @@
     public abstract void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
 
     /**
+     * Clear the blob data and metadata which is already removed from ZK.
+     * Implementation should just remove the blob as it is.
+     * This method should be used ONLY for syncing up blobs in Nimbus.
+     * @param key Key for the blob.
+     * @throws AuthorizationException
+     */
+    public abstract void clearRemovedBlob(String key) throws AuthorizationException;
```

Comment:
Is `BlobStore` a public API, which users can write their own implementations? If so, then this change would not be a backward compatible change for those users.